x/merkledb/trie_test.go

```diff
@@ -281,56 +271,34 @@ func Test_Trie_InsertAndRetrieve(t *testing.T) {
 	require.NoError(err)
 	require.NotNil(dbTrie)
 
-	trie := Trie(dbTrie)
-
 	// value hasn't been inserted so shouldn't exist
 	value, err := dbTrie.Get([]byte("key"))
 	require.ErrorIs(err, database.ErrNotFound)
 	require.Nil(value)
 
-	require.NoError(trie.Insert(context.Background(), []byte("key"), []byte("value")))
+	require.NoError(dbTrie.Put([]byte("key"), []byte("value")))
 
-	value, err = getNodeValue(trie, "key")
+	value, err = getNodeValue(dbTrie, "key")
 	require.NoError(err)
 	require.Equal([]byte("value"), value)
 }
 
-func Test_Trie_Overwrite(t *testing.T) {
```

Comment:
How come this got deleted?

Reply:
you can't overwrite if it is only writeable on creation

Reply:
It's possible to call `insert` multiple times with the same key. Namely, if the batch passed into `NewView` contains two `BatchOp` with the same key.